### PR TITLE
[5.6] Fix Request::get() to correctly retrieve merged inputs in a JSON request

### PR DESCRIPTION
**Pull request by Alymosul**
Currently in Laravel if you merged an input to an existing request and you try to retrieve it with the different methods on the Request class [get, only, input] they will all work as expected. However if this is a JSON request and you tried to merge an input and retrieve it through the `get()` method, it won't return the merged input, it always fetches the original inputs.

Scenario to produce the problem:

```
$payload = ['name' => 'Taylor'];
$request = Request::create('/', 'GET', [], [], [], ['CONTENT_TYPE' => 'application/json'], json_encode($payload));

$merge = ['age' => 26];
$request->merge($merge);

$request->get('age'); // Output: null
$request->input('age'); //Output: 26
```

### CODE REVIEW

**Comment by Alymosul** [comment]
I think we should have consistent behavior across all the methods that retrieves inputs, therefore input()/get()/all() should retrieve the correct inputs. Also this will not introduce a breaking change, since in the case of not having a JSON request we call the get() method implemented in the parent which is Symfony's Request class. It's not documented because it's not implemented it is inherited from Symfony but Laravel adds the JSON methods to the Request class...so maybe it was not expected that the get() to introduce a different behavior.

**Comment by vlakoff** [comment]
See https://github.com/laravel/ideas/issues/114, which hasn't been solved (yet).

tl;dr: It a symfony-inherited method, that we shouldn't alter or block. We best we could (and should) do would be to add a phpdoc telling to not use this `get()` method, but `input()` instead.

**Comment by taylorotwell** [comment]
Agree with prior comments.
